framework/src/meshgenerators/XYMeshLineCutter.C

```diff
@@ -0,0 +1,170 @@
+  if (!replicated_mesh_ptr)
+    paramError("input", "Input is not a replicated mesh, which is required");
+  if (*(replicated_mesh_ptr->elem_dimensions().begin()) != 2 ||
+      *(replicated_mesh_ptr->elem_dimensions().rbegin()) != 2)
+    paramError("input", "Only 2D meshes are supported.");
+
+  ReplicatedMesh & mesh = *replicated_mesh_ptr;
+
+  std::set<subdomain_id_type> subdomain_ids_set;
+  mesh.subdomain_ids(subdomain_ids_set);
+  const subdomain_id_type max_subdomain_id = *subdomain_ids_set.rbegin();
+  const subdomain_id_type block_id_to_remove = max_subdomain_id + 1;
+  const subdomain_id_type tri_subdomain_id_shift =
+      _tri_elem_subdomain_shift == Moose::INVALID_BLOCK_ID ? max_subdomain_id + 2
+                                                           : _tri_elem_subdomain_shift;
+
+  if (_cutting_type == CutType::CUT_ELEM_TRI)
+  {
+    try
+    {
+      MooseMeshXYCuttingUtils::lineRemoverCutElem(mesh,
+                                                  _cut_line_params,
+                                                  tri_subdomain_id_shift,
+                                                  _tri_elem_subdomain_name_suffix,
+                                                  block_id_to_remove,
+                                                  _new_boundary_id,
+                                                  _improve_tri_elements);
+    }
+    catch (MooseException & e)
+    {
+      if (((std::string)e.what()).compare("The new subdomain name already exists in the mesh.") ==
+          0)
+        paramError("tri_elem_subdomain_name_suffix",
+                   "The new subdomain name already exists in the mesh.");
+    }
+  }
+  else
+  {
+    try
+    {
+      MooseMeshXYCuttingUtils::lineRemoverMoveNode(mesh,
+                                                   _cut_line_params,
+                                                   block_id_to_remove,
+                                                   subdomain_ids_set,
+                                                   _new_boundary_id,
+                                                   _input_mesh_external_boundary_id,
+                                                   _other_boundaries_to_conform);
+    }
+    catch (MooseException & e)
+    {
+      if (((std::string)e.what())
+              .compare("The input mesh has degenerate quad element before trimming.") == 0)
+        paramError("input", "The input mesh has degenerate quad element before trimming.");
+      else if (((std::string)e.what())
+                   .compare("The new subdomain name already exists in the mesh.") == 0)
+        paramError("tri_elem_subdomain_name_suffix",
+                   "The new subdomain name already exists in the mesh.");
+    }
+    MooseMeshXYCuttingUtils::quasiTriElementsFixer(
+        mesh, subdomain_ids_set, tri_subdomain_id_shift, _tri_elem_subdomain_name_suffix);
+  }
+
+  mesh.prepare_for_use();
```

Comment:
let s remove that
it can trigger a repartitioning, and more often than not that could wait for the final mesh generation step

Reply:
well here it s replicated so maybe no repartitioning.

Reply:
for future reference, the proper thing to do here is to declare it as not prepared (set_not_prepared)